### DISCUSSION TITLE
How can I receive a File or Blob instance from frontend ?

### DISCUSSION THREAD

**Discussion by LXG19961206** [wailsapp/wails#4197] [Q&A]
### Is your feature request related to a problem? Please describe.

a simple code example here...

// js
let blob = new Blob(["hello world"], {type: "text/plain;charset=utf-8"});  
await ReceiveBlob(blob)

// wails go
func (rec *app) (blob any)  ReceiveBlob {
   ...
   // when I can this function, I got  `map[string]interface{}`, and the map has no key  , how can i receive it as a blob instance
}

### Describe the solution you'd like

please help me

**Comment by leaanthony** [comment]
Thanks for opening this.
What are you really trying to achieve with this? Sending a file from JS counter-intuitive for a Go program. There's probably a better way of doing it.

**Comment by LXG19961206** [comment]
> Thanks for opening this. What are you really trying to achieve with this? Sending a file from JS counter-intuitive for a Go program. There's probably a better way of doing it.

My business needs to use HTMLCanvas to dynamically draw data, then convert it into images and hand it over to the Wails Golang process,
Unable to directly call native functions of wails

**Comment by mjiulee** [comment]
cover to base64 data as string ,then send to go.

**Comment by mateothegreat** [comment]
The signature `blob any` is why you're getting `map[string]interface{}`.

You need to serialize your data and then send it to wails like:

```js
async function greet() {
  let blob = new Blob(["hello world"], { type: "text/plain;charset=utf-8" });
  // base64 encoded result:
  const result = btoa(await blob.text());

  Greet(result).then((result) => {
    greeting = result;
  });
}
```

```go
func (a *App) Greet(blob string) string {
...
}
```

![CleanShot 2024-05-18 at 00 46 41](https://github.com/wailsapp/wails/assets/11623940/98bd6ca4-a587-44c6-8a5e-50d2ac7f597c)
